msbuild/Xamarin.iOS.Tasks.Core/Tasks/DetectSdkLocationsTaskBase.cs

```diff
@@ -88,7 +88,9 @@ public abstract class DetectSdkLocationsTaskBase : Task
 
 		public override bool Execute ()
 		{
-			AppleSdkSettings.Init ();
+			if (!AppleSdkSettings.IsValid)
+				AppleSdkSettings.Init ();
```

Comment:
I would probably log both cases:
```csharp
Log.LogMessage (MessageImportance.Low, "Calling AppleSdkSettings.Init");
```
or:
```csharp
Log.LogMessage (MessageImportance.Low, "Skipping AppleSdkSettings.Init, already valid.");
```

You should also add a test that checks a build with no changes skips the `Init` call.

Something like this one:
https://github.com/xamarin/xamarin-macios/blob/4544fb8fcd987e53790c5bda5cd251a015a518d7/msbuild/tests/Xamarin.iOS.Tasks.Tests/ProjectsTests/XamarinForms.cs#L25-L28

You should be able to assert the "Skipping..." log message was printed.

Reply:
Also add some timings in the commit message that shows the savings for multiple projects. 👍